internal/backend/connections/connections.go

```diff
@@ -30,6 +30,10 @@ func (c *Connections) Create(ctx context.Context, conn sdktypes.Connection) (sdk
 		return sdktypes.InvalidConnectionID, err
 	}
 
+	if intg == nil {
```

Comment:
maybe it's better to handle this in integration interface? e.g. return sdk.ErrNotFound along with nil integration?

Reply:
there are some stuff that rely on it returning nil/invalid. we need to make a pass over all of these and settle on a convention.